snuba/web/rpc/v1/endpoint_time_series.py

```diff
@@ -0,0 +1,203 @@
+    result_timeseries: dict[tuple[str, str], TimeSeries] = {}
+
+    # create a mapping for each timeseries of timestamp: row to fill data points not returned in the query
+    result_timeseries_timestamp_to_row: defaultdict[
+        tuple[str, str], dict[int, Dict[str, Any]]
+    ] = defaultdict(dict)
+    query_duration = (
+        request.meta.end_timestamp.seconds - request.meta.start_timestamp.seconds
+    )
+    time_buckets = [
+        Timestamp(seconds=(request.meta.start_timestamp.seconds) + secs)
+        for secs in range(0, query_duration, request.granularity_secs)
+    ]
+
+    for row in data:
+        group_by_map = {}
+
+        for col_name, col_value in row.items():
+            if col_name in group_by_labels:
+                group_by_map[col_name] = col_value
+
+        group_by_key = "|".join([f"{k},{v}" for k, v in group_by_map.items()])
+        for col_name in aggregation_labels:
+            if not result_timeseries.get((group_by_key, col_name), None):
+                result_timeseries[(group_by_key, col_name)] = TimeSeries(
+                    group_by_attributes=group_by_map,
+                    label=col_name,
+                    buckets=time_buckets,
+                )
+            result_timeseries_timestamp_to_row[(group_by_key, col_name)][
+                int(datetime.fromisoformat(row["time"]).timestamp())
+            ] = row
+
+    # Go through every possible time bucket in the query, if there's row data for it, fill in its data
+    # otherwise put a dummy datapoint in
+
+    for bucket in time_buckets:
+        for timeseries_key, timeseries in result_timeseries.items():
+            row_data = result_timeseries_timestamp_to_row.get(timeseries_key, {}).get(
+                bucket.seconds
+            )
+            if not row_data:
+                timeseries.data_points.append(DataPoint(data=0, data_present=False))
+            else:
+                timeseries.data_points.append(
+                    DataPoint(data=row_data[timeseries.label], data_present=True)
+                )
+    return result_timeseries.values()
+
+
+def _build_query(request: TimeSeriesRequest) -> Query:
+    # TODO: This is hardcoded still
+    entity = Entity(
+        key=EntityKey("eap_spans"),
+        schema=get_entity(EntityKey("eap_spans")).get_data_model(),
+        sample=None,
+    )
+
+    aggregation_columns = [
+        SelectedExpression(
+            name=aggregation.label, expression=aggregation_to_expression(aggregation)
+        )
+        for aggregation in request.aggregations
+    ]
+
+    groupby_columns = [
+        SelectedExpression(
+            name=attr_key.name, expression=attribute_key_to_expression(attr_key)
+        )
+        for attr_key in request.group_by
+    ]
+
+    res = Query(
+        from_clause=entity,
+        selected_columns=[
+            SelectedExpression(name="time", expression=column("time", alias="time")),
```

Comment:
I would rather not have things depend on the TimeSeriesProcessor, but that is a personal opinion. I've been trying to deprecate that processor for a while.